pkg/operator/operator_test.go

```diff
@@ -281,3 +285,292 @@ func TestRunReport(t *testing.T) {
+				},
+			},
+			expectedReport: runReport{
+				degraded: &stateInfo{
+					reason:   "UpdatingFooFailed",
+					status:   "True",
+					messages: []string{"UpdatingFoo: Foo failed"},
+				},
+				available: &stateInfo{
+					reason:   "UpdatingFooFailed",
+					status:   "False",
+					messages: []string{"UpdatingFoo: Foo failed"},
+				},
+			},
+		},
+		{
+			name: "one failing UWM task with an unknown Degraded StateError",
+			taskGroupErrors: tasks.TaskGroupErrors{
+				tasks.TaskErr{
+					Err:        client.NewUnknownDegradedError("Foo failed"),
+					UWMRelated: true,
+					Name:       "UpdatingUserWorkloadFoo",
+				},
+			},
+			expectedReport: runReport{
+				degraded: &stateInfo{
+					reason:   "UpdatingUserWorkloadFooFailed",
+					status:   "Unknown",
+					messages: []string{"UpdatingUserWorkloadFoo: Foo failed"},
+				},
+				available: asExpected(client.TrueStatus),
+			},
+		},
+		{
+			name: "one failing Platform task with an Unavailable StateError",
+			taskGroupErrors: tasks.TaskGroupErrors{
+				tasks.TaskErr{
+					Err:        client.NewAvailabilityError("Foo failed"),
+					UWMRelated: false,
+					Name:       "UpdatingFoo",
+				},
+			},
+			expectedReport: runReport{
+				degraded: asExpected(client.FalseStatus),
+				available: &stateInfo{
+					reason:   "UpdatingFooFailed",
+					status:   "False",
+					messages: []string{"UpdatingFoo: Foo failed"},
+				},
+			},
+		},
+		{
+			name: "one failing Platform task with an Aggregate error",
+			taskGroupErrors: tasks.TaskGroupErrors{
+				tasks.TaskErr{
+					Err: apiutilerrors.NewAggregate(
+						[]error{
+							client.NewDegradedError("Foo failed bar"),
+							client.NewAvailabilityError("Foo failed baz"),
+						},
+					),
+					UWMRelated: false,
+					Name:       "UpdatingFoo",
+				},
+			},
+			expectedReport: runReport{
+				degraded: &stateInfo{
+					reason:   "UpdatingFooFailed",
+					status:   "True",
+					messages: []string{"UpdatingFoo: Foo failed bar"},
+				},
+				available: &stateInfo{
+					reason:   "UpdatingFooFailed",
+					status:   "False",
+					messages: []string{"UpdatingFoo: Foo failed baz"},
+				},
+			},
+		},
+		{
+			name: "multiple failing UWM tasks with generic errors",
+			taskGroupErrors: tasks.TaskGroupErrors{
+				tasks.TaskErr{
+					Err:        fmt.Errorf("Foo failed"),
+					UWMRelated: true,
+					Name:       "UpdatingUserWorkloadFoo",
+				},
+				tasks.TaskErr{
+					Err:        fmt.Errorf("Bar failed"),
+					UWMRelated: true,
+					Name:       "UpdatingUserWorkloadBar",
+				},
+			},
+			expectedReport: runReport{
+				degraded: &stateInfo{
+					reason:   "UserWorkloadTasksFailed",
+					status:   "True",
+					messages: []string{"UpdatingUserWorkloadFoo: Foo failed", "UpdatingUserWorkloadBar: Bar failed"},
+				},
+				available: &stateInfo{
+					reason:   "UserWorkloadTasksFailed",
+					status:   "False",
+					messages: []string{"UpdatingUserWorkloadFoo: Foo failed", "UpdatingUserWorkloadBar: Bar failed"},
```

Comment:
Added the task name to the `messages` to make it clearer in case of multiple tasks failing (also in case of 1 task to keep the code simple)